kotskinds/apis/kots/v1beta1/identityconfig_types.go

```diff
@@ -17,36 +17,75 @@ limitations under the License.
-	CACertPemBase64        string                `json:"caCertPemBase64,omitempty" yaml:"caCertPemBase64,omitempty"`
-	InsecureSkipTLSVerify  bool                  `json:"insecureSkipTLSVerify,omitempty" yaml:"insecureSkipTLSVerify,omitempty"`
-	Storage                Storage               `json:"storage,omitempty" yaml:"storage,omitempty"`
-	ClientID               string                `json:"clientID,omitempty" yaml:"clientID,omitempty"`
-	ClientSecret           string                `json:"clientSecret,omitempty" yaml:"clientSecret,omitempty"`
-	DexConnectors          DexConnectors         `json:"dexConnectors,omitempty" yaml:"dexConnectors,omitempty"`
+	Enabled                bool                    `json:"enabled" yaml:"enabled"`
+	DisablePasswordAuth    bool                    `json:"disablePasswordAuth,omitempty" yaml:"disablePasswordAuth,omitempty"`
+	Groups                 []IdentityConfigGroup   `json:"groups,omitempty" yaml:"groups,omitempty"`
+	IngressConfig          IngressConfigSpec       `json:"ingressConfig,omitempty" yaml:"ingressConfig,omitempty"`
+	AdminConsoleAddress    string                  `json:"adminConsoleAddress,omitempty" yaml:"adminConsoleAddress,omitempty"` // TODO (ethan): this does not belong here
+	IdentityServiceAddress string                  `json:"identityServiceAddress,omitempty" yaml:"identityServiceAddress,omitempty"`
+	CACertPemBase64        string                  `json:"caCertPemBase64,omitempty" yaml:"caCertPemBase64,omitempty"`
+	InsecureSkipTLSVerify  bool                    `json:"insecureSkipTLSVerify,omitempty" yaml:"insecureSkipTLSVerify,omitempty"`
+	Storage                Storage                 `json:"storage,omitempty" yaml:"storage,omitempty"`
+	ClientID               string                  `json:"clientID,omitempty" yaml:"clientID,omitempty"`
+	ClientSecret           *StringValueOrEncrypted `json:"clientSecret,omitempty" yaml:"clientSecret,omitempty"`
+	DexConnectors          DexConnectors           `json:"dexConnectors,omitempty" yaml:"dexConnectors,omitempty"`
+}
+
+type StringValueOrEncrypted struct {
+	Value          string `json:"value,omitempty" yaml:"value,omitempty"`
+	ValueEncrypted string `json:"valueEncrypted,omitempty" yaml:"valueEncrypted,omitempty"`
+}
+
+func NewStringValueOrEncrypted(value string, cipher crypto.AESCipher) *StringValueOrEncrypted {
+	v := &StringValueOrEncrypted{Value: value}
+	v.EncryptValue(cipher)
+	return v
+}
+
+func (v *StringValueOrEncrypted) GetValue(cipher crypto.AESCipher) (string, error) {
+	if v == nil {
+		return "", nil
+	}
+	if v.ValueEncrypted != "" {
+		b, err := base64.StdEncoding.DecodeString(v.ValueEncrypted)
+		if err != nil {
+			return "", errors.Wrap(err, "failed to base64 decode")
+		}
+		result, err := cipher.Decrypt(b)
+		return string(result), errors.Wrap(err, "failed to decrypt")
+	}
+	return v.Value, nil
+}
+
+func (v *StringValueOrEncrypted) EncryptValue(cipher crypto.AESCipher) {
+	if v.ValueEncrypted != "" && v.Value == "" {
```

Comment:
what if both are empty?

Reply:
then i guess its an empty string

Reply:
ok, just wanted to make sure `Encrypt` won't fail